### PR TITLE
[SourceKit/SwiftLang] Keep sourcekitd response alive for variant lifetime

### PR DESCRIPTION
**Pull request by nathawes**
This PR contains two changes to SwiftLang, the Swift layer exposing SourceKit in Swift:

1) Make each Variant keep a strong reference to its `SourceKitdResponse` context. This is needed because `sourcekitd_variant_t` is only safe to use while the `sourcekitd_response_t` it was retrieved from (wrapped by `SourceKitdResponse`) is still alive.
2) Expose the new `Data` variant SourceKit APIs that were added to support returning the Syntax tree in a binary format in SwiftLang.

### CODE REVIEW

**Comment by nathawes** [comment]
@swift-ci please test

**Review by nkcsgexi — APPROVED** [review]
LGTM, thank you for fixing it!

**Comment by nkcsgexi** [comment]
Could we also cherry-pick this to swift-5.0-branch afterwards? It's useful to verify syntax tree binary format in a toolchain.

File: tools/SourceKit/tools/swift-lang/SourceKitdResponse.swift [hunk 1]
```diff
@@ -47,12 +53,21 @@ public class SourceKitdResponse: CustomStringConvertible {
...
     }
 
     public func getDictionary(_ key: SourceKitdUID) -> Dictionary {
       let value = sourcekitd_variant_dictionary_get_value(dict, key.uid)
-      return Dictionary(dict: value)
+      return Dictionary(dict: value, context: context)
+    }
+
+    public func getData(_ key: SourceKitdUID) -> Data {
+      let value = sourcekitd_variant_dictionary_get_value(dict, key.uid)
```
Comment (nkcsgexi) [inline tools/SourceKit/tools/swift-lang/SourceKitdResponse.swift]: why not using `sourcekitd_variant_data_get_ptr ` here?

Comment (nkcsgexi) [reply tools/SourceKit/tools/swift-lang/SourceKitdResponse.swift]: Never mind, i misread the change 😄

**Pull request merged**